docs/src/test/java/docs/http/javadsl/server/directives/HeaderDirectivesExamplesTest.java

```diff
@@ -36,87 +39,113 @@ public void testHeaderValue() {
     };
 
     final Route route = headerValue(extractHostPort, host ->
-        complete("The port was " + host.port())
+      complete("The port was " + host.port())
     );
 
     // tests:
     testRoute(route).run(HttpRequest.GET("/").addHeader(Host.create("example.com", 5043)))
-                    .assertEntity("The port was 5043");
+      .assertEntity("The port was 5043");
 
     testRoute(route).run(HttpRequest.GET("/"))
-                    .assertStatusCode(StatusCodes.NOT_FOUND)
-                    .assertEntity("The requested resource could not be found.");
+      .assertStatusCode(StatusCodes.NOT_FOUND)
+      .assertEntity("The requested resource could not be found.");
     //#headerValue
   }
 
   @Test
+  public void testHeaderValueWithDefault() {
+    //#headerValue-with-default
+    final Function<HttpHeader, Optional<String>> extractExampleHeader = header -> {
+      if (header.name().equalsIgnoreCase("exampleHeader")) {
```

Comment:
You could also use `header.is("exampleheader")` (`is` requires a lower case name to compare against). Maybe it would make sense to not use camel case for a header name but something more "realistic" like `X-Example-Header` and then `header.is("x-example-header")`.